File: Python/quadratic_eq.py
from math import sqrt

print("Considere la expresión ax^2+bx+c=0")

while(True):
    try:
        a = float(input("Introduzca el coeficiente de la variable cuadratica 'a': "))
        break
    except:
        print("Introduce un número real")
while(True):
    try:
        b = float(input("Introduzca el coeficiente de la variable lineal 'b' : "))
        break
    except:
        print("Introduce un número real")
while(True):
    try:
        c = float(input("Introduzca el coeficiente de la constante 'c' : "))
        break
    except:
        print("Introduce un número real")

print(f"La expresión es ({a}) X^2 + ({b}) X + {c} = 0")

if b**2 - 4*a*c >= 0:
    if a != 0:
        x1 = (-b + sqrt((b**2) -4*a*c))/(2*a)
        x2 = (-b - sqrt((b**2) -4*a*c))/(2*a)
        print(f"La solución positiva x1 = {x1} \nLa solución negativa x2 = {x2}")
    else:
        if b != 0:
            if c == 0:
                print("La solución es x = 0")
            else:
                x = -c/b
                print(f"La solución es {x}")
        else:
            if c != 0:
                print("La ecuación no tiene solución")
            else: 
                print("La ecuación tiene infinitas soluciones")
else:
    print("La ecuacion no tiene solución perteneciente a los Reales")

File: C++/quadratic_eq.cpp
#include <stdio.h>
#include <math.h>
//#include <conio.h>

int main()
{
    float a,b,c,x1,x2,x;

    printf("Considera la expresión ax^2 + bx + c = 0 \n Introduce los valores de a, b y c: \n");

    scanf("%f%f%f",&a,&b,&c);

    printf("La expresión es %0.2f x^2 + %0.2f + %0.2f = 0 \n",a,b,c);

    if(b*b - 4*a*c >= 0)
    {
        if(a != 0)
        {
            x1 = (-b + sqrt(b*b - 4*a*c))/(2*a);
            x2 = (-b - sqrt(b*b - 4*a*c))/(2*a);
            printf("La solucion positiva es x = %0.2f \nLa solucion negativa es x = %0.2f \n",x1,x2);
        }
        else
        {
            if(b != 0)
            {
                if(c == 0)
                {
                    printf("La solución es x = 0 \n");
                }
                else
                {
                    x = -c/b;
                    printf("La solución es x = %0.2f \n", x);
                }
            }
            else
            {
                if(c != 0)
                {
                    printf("La ecuación no tiene solución \n");
                }
                else
                {
                    printf("La ecuación no tiene sentido \n");
                }

            }

        }
    }
    else
    {
        printf("La ecuación no tiene solución dentro de los Reales \n");
    }
    //getch();
}
